### DISCUSSION TITLE
OpenVideoMixProjector / OpenSourceProjector vs OpenProjector/ closemonitors / CloseProjector

### DISCUSSION THREAD

**Discussion by henry-delaruelle** [Palakis/obs-websocket#1007] [Q&A]
I am working on a C# library to use the new 5.0 obs-websocket, most of it works, but i run in to problems when i try to open projectors,
i can not use it in 5.0 and it does not work via the 4.x compat plugin.

I used the OpenProjector command to open a preview monitor on a second/third monitor connected to the computer

In version 5 this command is replace by two commands : OpenVideoMixProjector / OpenSourceProjector, but i can not get it to work. 

with a  -1 as monitorIndex the projectorGeometry does not give the expected result (i get a small windows, not the size i expect) with a monitorIndex valye > -1 i would expect a fullscreen preview but nothing happends.

the closemonitors command is replaced by CloseProjector, but this is not implemented yet ?

i have some questions:

1.  Is the projectorGeometry string in qt5 different than in qt6? and if so what is changed ?
2. What should i use a monitorIndex, in the documentation a see a GetMonitorList  command, but i don't see an event that returns a MonitorList.
3. any idea when the CloseProjector request will be implemented. I use the c# library to control obs from an android app, and when the projector functions work, i can move to OBS 5.01, please advice

**Comment by tt2468** [comment]
1. I honestly have no clue. Possibly? You'd have to look up the Qt docs for that.
2. The `GetMonitorList` request returns an ordered list of monitors, and you use the index to open projectors, as described.
3. We have never had a `closemonitors` request, I have no idea where you would have seen that before. There is no estimate for when CloseProjector will be implemented, as the whole projector system really needs a rewrite.

**Reply by henry-delaruelle** [reply]
1. I could not find the information i needed in the documentation, fortunately the ini file of obs contained some projectorGeometry items, i could use these to find out the structure.
2. Thanks, that was what i needed !!
3. Sorry, I wrote my own, it would be helpfull if it was available as a command
